openfl-tutorials/interactive_api/PyTorch_Lightning_MNIST_GAN/envoy/mnist_shard_descriptor.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Mnist Shard Descriptor."""
+
+import logging
+from typing import List
+
+from torchvision import datasets
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDataset
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+
+logger = logging.getLogger(__name__)
+
+
+class MnistShardDataset(ShardDataset):
+    """Mnist Shard dataset class."""
+
+    def __init__(self, x, y, data_type, rank=1, worldsize=1):
+        """Initialize Mnist shard Dataset."""
+        self.data_type = data_type
+        self.rank = rank
+        self.worldsize = worldsize
+
+        self.x = x[self.rank - 1::self.worldsize]
+        self.y = y[self.rank - 1::self.worldsize]
+
+    def __getitem__(self, index: int):
+        """Return an item by the index."""
+        return self.x[index], self.y[index]
+
+    def __len__(self):
+        """Return the len of the dataset."""
+        return len(self.x)
+
+
+class MnistShardDescriptor(ShardDescriptor):
+    """Mnist Shard descriptor class."""
+
+    def __init__(
+            self,
+            rank_worldsize: str = '1, 1',
+            **kwargs
+    ):
+        """Initialize MnistShardDescriptor."""
+        self.rank, self.worldsize = tuple(int(num) for num in rank_worldsize.split(','))
+        (x_train, y_train), (x_val, y_val) = self.download_data()
+        self.data_by_type = {
+            'train': (x_train, y_train),
+            'val': (x_val, y_val)
+        }
+
+    def get_shard_dataset_types(self) -> List[str]:
```

Comment:
I think returned type annotation should be different
```suggestion
    def get_shard_dataset_types(self) -> List[Dict[str, Any]]:
```